src/components/validation/resources/event.rs

```diff
@@ -36,17 +38,26 @@ pub enum RawTestEvent {
 }
 
 #[derive(Clone, Debug, Deserialize)]
-#[serde(untagged)]
+#[serde(rename_all = "snake_case")]
 pub enum EventData {
-    /// A log event.
+    /// A simple log event.
     Log(String),
+    /// A log event built from key-value pairs
+    LogBuilder(HashMap<String, String>),
 }
 
 impl EventData {
     /// Converts this event data into an `Event`.
     pub fn into_event(self) -> Event {
         match self {
             Self::Log(message) => Event::Log(LogEvent::from_bytes_legacy(&message.into())),
+            Self::LogBuilder(data) => {
+                let mut log_event = LogEvent::default();
+                for (k, v) in data {
+                    log_event.parse_path_and_insert(k, v).unwrap();
```

Comment:
Is it safe to do a raw `unwrap()` here?

Reply:
this is all code in a test framework so its much safer than production code. But I'll change this to an `expect` and give a more helpful error message.